bftengine/tests/bcstatetransfer/source_selector_test.cpp

```diff
@@ -67,39 +68,39 @@ TEST_F(SourceSelectorTestFixture, should_change_source_after_construction) {
   // Make sure that the current replica is "NO_REPLICA".
   // When there is no current replica the selector should select an initial replica as source.
   ASSERT_EQ(source_selector.currentReplica(), NO_REPLICA);
-  ASSERT_TRUE(source_selector.shouldReplaceSource(kSampleCurrentTimeMs, false));
+  ASSERT_EQ(source_selector.shouldReplaceSource(kSampleCurrentTimeMs, false, false), SourceReplaceType::IMMEDIATE);
 }
 
 TEST_F(SourceSelectorTestFixture, should_change_source_when_bad_data_is_received_from_the_current_source) {
   // Select a source
-  source_selector.updateSource(kSampleCurrentTimeMs);
+  source_selector.updateSource(kSampleCurrentTimeMs, SourceReplaceType::IMMEDIATE);
   // Ensuring that the change of the selected source is not caused by a timeout is
   // done by intentionally passing a value than is not much bigger than the one stated as current time during the
   // selection process.
-  ASSERT_TRUE(source_selector.shouldReplaceSource(kSmallDeltaTime, true));
+  ASSERT_EQ(source_selector.shouldReplaceSource(kSmallDeltaTime, true, false), SourceReplaceType::IMMEDIATE);
 }
 
 TEST_F(SourceSelectorTestFixture, should_change_source_when_the_replacement_time_has_elapsed) {
-  source_selector.updateSource(kSampleCurrentTimeMs);
-  ASSERT_TRUE(source_selector.shouldReplaceSource(kBigDeltaTime, false));
+  source_selector.updateSource(kSampleCurrentTimeMs, SourceReplaceType::IMMEDIATE);
+  ASSERT_EQ(source_selector.shouldReplaceSource(kBigDeltaTime, false, true), SourceReplaceType::GRACEFUL);
 }
 
 TEST_F(SourceSelectorTestFixture, source_should_not_be_changed_when_working_and_replacement_time_has_not_elapsed_yet) {
-  source_selector.updateSource(kSampleCurrentTimeMs);
-  ASSERT_FALSE(source_selector.shouldReplaceSource(kSmallDeltaTime, false));
+  source_selector.updateSource(kSampleCurrentTimeMs, SourceReplaceType::IMMEDIATE);
+  ASSERT_EQ(source_selector.shouldReplaceSource(kSmallDeltaTime, false, false), SourceReplaceType::DONOT);
 }
 
 // Remove current replica
 TEST_F(SourceSelectorTestFixture, remove_current_replica_sets_the_current_replica_to_no_replica) {
   ASSERT_EQ(source_selector.currentReplica(), NO_REPLICA);
-  source_selector.updateSource(kSampleCurrentTimeMs);
+  source_selector.updateSource(kSampleCurrentTimeMs, SourceReplaceType::IMMEDIATE);
```

Comment:
Hi @mdarade , I though you are the one recommended (and I agree) to keep the source in the case of periodic replacment.
Did you change you mind?
Also, how does this relate to the above test?

Reply:
SS doesn't chose source in round robin fashion. SS choses source randomly from preferredList.
If we don't remove source from preferredList, it may get chosen again.

This invalidates the need of passing extra info to updateSource() which has doesn't need SourceReplaceType anymore.